src/py_zfs_resource.c

```diff
@@ -129,6 +115,20 @@ PyObject *py_zfs_resource_get_properties(PyObject *self,
 		return NULL;
 	}
 
+	if (res->is_simple) {
+		/*
+		 * We have simple handle that lacks property information.
+		 * This means we _must_ refresh properties before
+		 * generating python object
+		 */
+		Py_BEGIN_ALLOW_THREADS
+		// This call does not touch ZFS errno and
```

Comment:
Do we need to aquire lock here and below using `PY_ZFS_LOCK` here on libzfs handle? Internally libzfs handle will be used in `get_stats_ioctl` and while allocating memory for nvlists.